File: src/DevProjects.Core/Services/SessionEndDetector.cs
namespace DevProjects.Core.Services;

/// <summary>Pure diff of running-session directory snapshots: which projects just ended.</summary>
public static class SessionEndDetector
{
    public static IEnumerable<string> Ended(IReadOnlySet<string> previous, IReadOnlySet<string> current) =>
        previous.Where(p => !current.Contains(p));
}

File: electron/src/core/claude/sessionEndDetector.ts
/** Pure diff of running-session snapshots: which projects just ended. */

/**
 * Returns the paths present in previous but absent in current (case-insensitive).
 */
export function ended(
  previous: ReadonlySet<string>,
  current: ReadonlySet<string>,
): string[] {
  const lowerCurrent = new Set([...current].map((s) => s.toLowerCase()))
  return [...previous].filter((p) => !lowerCurrent.has(p.toLowerCase()))
}
